iroh-net/src/netcheck/reportgen.rs

```diff
@@ -564,9 +572,7 @@ impl Actor {
     ///     failure permanent.  Probes in a probe set are essentially retries.
     ///   - Once there are [`ProbeReport`]s from enough regions, all remaining probes are
     ///     aborted.  That is, the main actor loop stops polling them.
-    async fn prepare_probes_task(
-        &mut self,
-    ) -> Result<FuturesUnordered<Pin<Box<impl Future<Output = Result<ProbeReport>>>>>> {
+    async fn prepare_probes_task(&mut self) -> Result<JoinSet<Result<ProbeReport>>> {
```

Comment:
This no longer prepares a future but will spawn the futures on tasks and they'll start running right away.  Maybe rename the function to `start_probe_tasks` or `spawn_probe_tasks` and update the first line of the docstring as well?

Reply:
renamed